frontend/src/components/types/AuctionWithProductInfo.ts

```diff
@@ -0,0 +1,14 @@
+import { ProductInterface } from "./ProductInterface";
+import { Bid } from "./Bid";
+export interface AuctionWithProductInfo {
+    profileID: string;
+    __v: number;
+    _id: string;
+    amount: number;
+    bidHistory: Bid[];
+    startDate: Date;
+    endDate: Date;
+    likes: number;
+    price: number;
+    productID: ProductInterface;
```

Comment:
suggestion: maybe you can find a way of renaming `productID` to product when populating the field with mongoose.

`productId` sounds like a string/number type of field, but actually contains an object.

Maybe something like this would work? 👉 https://stackoverflow.com/a/46930284